src/spec/create_extension_spec.py

```diff
@@ -0,0 +1,531 @@
+# -*- coding: utf-8 -*-
+import os.path
+
+from pynwb.spec import NWBAttributeSpec, NWBDatasetSpec, NWBGroupSpec, NWBLinkSpec, NWBNamespaceBuilder, export_spec
+
+
+def main():
+    ns_builder = NWBNamespaceBuilder(
+        name="""ndx-extracellular-channels""",
+        version="""0.1.0""",
+        doc="""NWB extension for storing extracellular probe and channels metadata""",
+        author=[
+            "Alessio Buccino",
+            "Kyu Hyun Lee",
+            "Ramon Heberto Mayorquin",
+            "Cody Baker",
+            "Matt Avaylon",
+            "Ryan Ly",
+            "Ben Dichter",
+            "Oliver Ruebel",
+            "Geeling Chau",
+        ],
+        contact=[
+            "alessio.buccino@alleninstitute.org",
+            "kyuhyun.lee@ucsf.edu",
+            "ramon.mayorquin@catalystneuro.com",
+            "cody.baker@catalystneuro.com",
+            "mavaylon@lbl.gov",
+            "rly@lbl.gov",
+            "ben.dichter@catalystneuro.com",
+            "oruebel@lbl.gov",
+            "gchau@caltech.edu",
+        ],
+    )
+    ns_builder.include_namespace("core")
+
+    contacts_table = NWBGroupSpec(
+        neurodata_type_def="ContactsTable",
+        neurodata_type_inc="DynamicTable",
+        doc="Metadata about the contacts of a probe, compatible with the ProbeInterface specification.",
+        default_name="contacts_table",
+        datasets=[
+            NWBDatasetSpec(
+                name="relative_position_in_mm",
+                neurodata_type_inc="VectorData",
+                doc="Relative position of the contact in millimeters, relative to `reference`.",
+                dtype="float",
+                dims=[["num_contacts", "x, y"], ["num_contacts", "x, y, z"]],
+                shape=[[None, 2], [None, 3]],
+                attributes=[
+                    NWBAttributeSpec(
+                        name="reference",
+                        doc=(
+                            "Reference point for the relative position coordinates and information about the "
+                            "coordinate system used, e.g., which direction is positive in the x direction "
+                            "(first element), which direction is positive in the y direction (second element), etc."
+                        ),
+                        dtype="text",
+                        required=False,
+                    )
+                ],
+            ),
+            NWBDatasetSpec(
+                name="contact_id",  # id is already used by DynamicTable
+                neurodata_type_inc="VectorData",
+                doc="Unique ID of the contact",
+                dtype="text",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
```

Comment:
I think there is another knot here.

If I remember well one of the purposes of separating Probe from ProbeModel is so the latter represents the catalogue version of the Probe which is an instance in the experimental setup.

The contacts table falls firmly on the catalogue because we are assuming that Probes come with a consisten geometry from the catalogue.

However, the mapping from the contacts to the channels here is clearly a property of the experimental setup so I don't think it makes a lot of conceptual sense with the previous intent.

That said, I understand that this is for the round-trip to work and we practicality beats purity so I think is fine.

To be honest, I don't really like the way that the wiring is managed in the interface between probe interface and spikeinterface. I feel that the mapping should be done from the recording to the probe and not the other way around. Plus, I think that it should be done from the channel ids (in spikeinterface) to the contacts ids (in probeinterface) instead of using a mapping from indices to indices which is an implementation detail of the data structures in both data models. 

I have not had the time to pitch these concents to Sam and Alessio but I think I should do this soon.

Reply:
The headstage (or data acquisition system) has a particular way it maps channels (0 to N-1) to contacts. This mapping is a property of the headstage. We don't really represent the headstage in NWB except with properties like filters in the `ChannelsTable` (or ElectrodesTable - see #3).

I think this column was introduced by @alejoe91 and @khl02007 in the initial iteration of ndx-probeinterface. I agree that it doesn't really belong on the `ProbeModel` and is more appropriate on `Probe` or `ChannelsTable`. Should we remove this for now? We can always add it back later once we are happy with it.

Reply:
No, I am not confused about what it is. I understand that there is a mapping from the DAQ/Controller to the probe (that sometimes passess through a headstage). We need that mapping and in probeinterface case is done through that field.

But I think that in our conceptual model it does not make sense to put the mapping there. So yes, I think we should remove it. We already have `contact` as a column in the channels table that is taking care of that in what I think is a conceptually cleaner way.

Reply:
> We already have `contact` as a column in the channels table that is taking care of that in what I think is a conceptually cleaner way.

Yeah, I think that is the better way at least in our model. 

OK will do.